Classes/ShapeGenerator.lua

```diff
@@ -0,0 +1,435 @@
+
+-- ShapeGenerator.lua
+
+-- Capable of generating shapes in BlockAreas. Either a predefined shape like a cylinder, or a shape from a mathematical formula.
+-- Only for a shape made using a mathematical formula the constructor has to be used. For other shapes there are static functions to create the shape in a blockarea.
+
+
+
+
+
+cShapeGenerator = {}
+
+
+
+
+
+-- The coordinates around a single point
+cShapeGenerator.m_Coords =
+{
+	Vector3f(1, 0, 0), Vector3f(-1, 0, 0), -- X coords
+	Vector3f(0, 1, 0), Vector3f(0, -1, 0), -- Y coords
+	Vector3f(0, 0, 1), Vector3f(0, 0, -1), -- Z coords
+}
+
+
+
+
+
+-- Handler that makes the structure hollow
+cShapeGenerator.m_HollowHandler = function(a_ShapeGenerator, a_BlockArea, a_BlockPos)
+	-- Check around the block coordinates if it has at least one single position that doesn't get set
+	for Idx, Coord in ipairs(cShapeGenerator.m_Coords) do
+		local CoordAround = a_BlockPos + Coord
+		local DoSet = a_ShapeGenerator:GetBlockInfoFromFormula(CoordAround)
+		if (not DoSet) then
+			-- Empty spot around the block.
+			return true
+		end
+	end --  /for Coords
+	return false
+end
+
+
+
+
+
+-- Handler that makes the structure solid. 
+cShapeGenerator.m_SolidHandler = function(a_ShapeGenerator, a_BlockArea, a_BlockPos)
+	return true
+end
+
+
+
+
+
+-- Creates a new cShapeGenerator object.
+-- a_Zero and a_Unit are Vector3f vectors used to calculate a scaled vector3f. The formula will use the scaled vector as x, y and z values.
+-- a_BlockTable are the blocks to make the shape out of.
+-- a_Expression is a cExpression object that compiles the formula into a safe function.
```

Comment:
"Compiles", or "is expected to have compiled"

Also, the function should probably check the parameters and assert if they don't match expectations.